plugins/kotlin-dataframe/testData/box/groupBy_refine.kt

```diff
@@ -0,0 +1,12 @@
+import org.jetbrains.kotlinx.dataframe.*
+import org.jetbrains.kotlinx.dataframe.annotations.*
+import org.jetbrains.kotlinx.dataframe.api.*
+import org.jetbrains.kotlinx.dataframe.io.*
+
+
+fun box(): String {
+    val df = dataFrameOf("a", "b", "c")(1,2,3)
+    val df1 = df.groupBy { a }
+    df1.keys.compareSchemas(strict = true)
```

Comment:
what does this test achieve actually?

Reply:
compareSchemas calls df.compileTimeSchema (=MarkersExtractor.get()), this function creates a DataFrameSchema from class declaration. In case with plugin, we expect that it will be a generated schema. Then it compares it to actual runtime schema.
So, previously this test would fail, because by default groupBy returns GroupBy<T, T>, so compileTimeSchema for keys
a: Int
b: Int
c: Int
and it's different from runtime
a: Int 
In short, this is the way to make sure there's no extra columns in the schema